### PR TITLE
site: add search function

### LINKED ISSUE #380: Add Search Capability to Documentation
**What would you like to be added**:
A search functionality to the project's documentation.

**Why is this needed**:
While navigating through the documentation of the current project, I found it challenging to locate specific information, especially when the content is extensive. A search feature would significantly enhance the user experience, allowing users to quickly find the information they're looking for.

### PR DESCRIPTION
**Pull request by Xieql**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
site: add search function

**Which issue(s) this PR fixes**:
Fixes #380

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
site: now user can search in website https://kurator.dev/
```
image

image

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
goog work

/lgtm

File: docs/config.toml [hunk 1]
```diff
@@ -60,6 +60,17 @@ path = "github.com/google/docsy/dependencies"
 
 [params]
 copyright = "The Kurator Authors"
+  
+# Google Custom Search Engine ID. Remove or comment out to disable search.
+#gcs_engine_id = "d72aa9b2712488cc3"
```
Comment (hzxuzhonghu) [inline docs/config.toml:65]: can we remove it

Comment (Xieql) [reply docs/config.toml:65]: ok

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**